hollow/src/main/java/com/netflix/hollow/tools/patch/delta/HollowStateDeltaPatcher.java

```diff
@@ -114,6 +115,7 @@ public void prepareInitialTransition() {
     public void prepareFinalTransition() {
         writeEngine.prepareForNextCycle();
         writeEngine.overrideNextStateRandomizedTag(to.getCurrentRandomizedTag());
+        writeEngine.addHeaderTags(to.getHeaderTags());
```

Comment:
The second transition now also populates header tags corresponding to the final state